procServ.cc

```diff
@@ -475,38 +475,44 @@ int main(int argc,char * argv[])
     memset(&sig, 0, sizeof(sig));
 
     PRINTF("Installing signal handlers\n");
-    
+
     // SIGPIPE, SIGTERM and SIGHUP will be handled in the main loop
     // with the assistance of pselect. This means that we have them
     // blocked outside of pselect call, but unblocked atomically
     // within pselect. Each time pselect returns, we safely check if
     // any of the signals were received.
-    
-    // Block the signals that we bill be handling in the main loop.
+
+    // Block the signals that we will be handling in the main loop
+    // and re-enable them during pselect.
     // At the same time, retrieve the original signal mask before
-    // blocking, to be passed to pselect.
+    // blocking, to be passed to the child processes.
+
     sigset_t sigset_block;
+    sigset_t sigset_child;
     sigset_t sigset_pselect;
     sigemptyset(&sigset_block);
     sigaddset(&sigset_block, SIGPIPE);
     sigaddset(&sigset_block, SIGTERM);
     sigaddset(&sigset_block, SIGHUP);
-    sigprocmask(SIG_BLOCK, &sigset_block, &sigset_pselect);
-    
+    sigaddset(&sigset_block, SIGXFSZ);
+    if (inFgMode) {
+        sigaddset(&sigset_block, SIGINT);
+        sigaddset(&sigset_block, SIGQUIT);
+    }
```

Comment:
This unblocks some more signals from the child, because the `SIG_IGN` disposition used in `sigaction` previously would still apply to the child processes, even if the sigmask was restored.

Would be nice if the commit message called out that nuance.